app/controllers/UrlController.php

```diff
@@ -9,188 +9,200 @@
-            'roles' => ['urlManage'],
-          ],
-          [
-            'allow' => true,
-            'actions' => ['index', 'update'],
-            'roles' => ['urlManage'],
-          ],
-        ],
-      ],
-    ];
-  }
-
-  /**
-   * Lists all NixShortUrls models.
-   * @return mixed
-   * @throws \yii\base\InvalidParamException
-   */
-  public function actionIndex()
-  {
-      $searchModel = new NixShortUrlsSearch();
-      $dataProvider = $searchModel->search(Yii::$app->request->getQueryParams());
-
-      return $this->render('index', [
-          'dataProvider' => $dataProvider,
-          'searchModel' => $searchModel,
-      ]);
-  }
-
-  /**
-   * Updates an existing NixShortUrls model.
-   * If update is successful, the browser will be redirected to the 'view' page.
-   * @param integer $id
-   * @return mixed
-   * @throws \yii\base\InvalidParamException
-   */
-  public function actionUpdate($id)
-  {
-    $model = $this->findModel($id);
-
-    if ($model->load(Yii::$app->request->post()) && $model->save()) {
-      return $this->redirect(['index']);
+    /**
+     * @return array
+     */
+    public function behaviors(): array
+    {
+        return [
+            'access' => [
+                'class' => AccessControl::className(),
+                'only' => ['index', 'view', 'update', 'delete', 'add'],
+                'rules' => [
+                    [
+                        'allow' => true,
+                        'actions' => ['add'],
+                        'verbs' => ['POST'],
+                        'matchCallback' => function () {
+                            return !Yii::$app->user->isGuest ? Yii::$app->user->can('urlCreate') : true;
+                        }
+                    ],
+                    [
+                        'allow' => true,
+                        'actions' => ['delete'],
+                        'verbs' => ['POST'],
+                        'roles' => ['urlManage'],
+                    ],
+                    [
+                        'allow' => true,
+                        'actions' => ['index', 'update'],
+                        'roles' => ['urlManage'],
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Lists all NixShortUrls models.
+     * @return string
+     */
+    public function actionIndex(): string
+    {
+        $searchModel = new NixShortUrlsSearch();
+        $dataProvider = $searchModel->search(Yii::$app->request->getQueryParams());
+
+        return $this->render(
+            'index',
+            [
+                'dataProvider' => $dataProvider,
+                'searchModel' => $searchModel,
+            ]
+        );
+    }
+
+    /**
+     * Updates an existing NixShortUrls model.
+     * If update is successful, the browser will be redirected to the 'view' page.
+     *
+     * @param integer $id
+     *
+     * @return string
+     */
+    public function actionUpdate($id): string
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://app.codacy.com/app/gaalferov/yii2-minify-url/pullRequest?prid=4708268)